src/amo/utils/addonDownload.ts

```diff
@@ -4,14 +4,61 @@ import * as vscode from "vscode";
 
 import constants from "../../config/config";
 
-export async function downloadAddon(fileId: string, path: string) {
+async function fetchDownloadFile(fileId: string) {
   const url = `${constants.downloadBaseURL}${fileId}`;
   const response = await fetch(url);
-
-  if (response.ok) {
-    const dest = fs.createWriteStream(path, { flags: "w" });
-    dest.write(await response.buffer());
-    dest.close();
-    vscode.window.showInformationMessage("Download complete");
+  if (!response.ok) {
+    await vscode.window
+      .showErrorMessage(
+        `(Status ${response.status}): Could not fetch download file.`,
+        { modal: true },
+        { title: "Try Again" },
+        { title: "Fetch New Addon" }
+      )
+      .then(async (action) => {
+        if (action?.title === "Try Again") {
+          return await fetchDownloadFile(fileId);
+        } else if (action?.title === "Fetch New Addon") {
+          vscode.commands.executeCommand("assay.get");
+          throw new Error("Process restarted");
```

Comment:
this looks like it would execute the command, and then throw an error anyway?  If that's not what's happening can you add a code comment here?

Reply:
We talked about this a while ago, but it throws an error to let the console know what happened, which ends the current fetch command and starts a new one. Without it, two commands would run in parallel.

Reply:
okay, that makes sense - can you add a brief comment in the code for future code maintainers.